### ISSUE TITLE
Input configuration for surfaces

### ISSUE THREAD

**Issue by HansKallekleiv** [equinor/webviz-subsurface#82]
Relates to #69 

In order to facilitate easy selection and visualization of surface data from an ensemble some functionality is needed to show which data is available.

Work is currently ongoing to resolve this (@perolavsvendsen) , but until then we need a band-aid solution that will allow users to visualise these data. Ideally we should be able to make something that later can be modified.

Below is a suggestion using a user-provided configuration file and a ui widget to select data. 

The suggestion has many limitations and  **is just meant to get the discussion started**

@jcrivenaes
@anders-kiaer 
@berland

Any feedback is welcome.

### How data is stored
Currently surface is stored as files in an ensemble on the following relative path
`share/results/maps`
Individual surface grids are stored within that path as:
`surface_name--surface_property--surface_timestep.gri`

The name components can be described as:
**Surface grid** a 3-dimensional collection of points, x, y, z, where x,y are coordinates while z is an arbritrary property.

**Surface name** The name of the entity that the surface represents. This could be e.g. a seismic interpreted reflector,  a geological interpreted isochore or a fluid contact.

**Surface property** The z-value attribute of the surface grid. This could be e.g. depth values, time values, mean porosity or saturation.

**Surface timestep** A surface property can have a time dimension, e.g. pressure changes over time

- *Surface names* and *Surface timesteps* must be ordered.
- Different *surface properties* can have different lists of names and timesteps
- Different *surface properties* does not have to be ordered

A simple way to display this information can be:
```yaml

#Configuration file for surfaces
#surface_property:
#  surface_names:
#    - some name
#    - another name
#  surface_timestep:
#    - some dat
#    - another date

average_permx:
  names: 
    - all
    - upperreek
    - midreek
    - lowerreek

average_pressure:
  names:
    - all
    - upperreek
    - midreek
    - lowerreek
  dates:
    - 20000101
    - 20010601
    - 20030101
    - 20010601_20000101
    - 20030101_20000101
    - 20030101_20010601

ds_extracted_horizons:
  names:
    - topupperreek
    - topmidreek
    - toplowerreek
    - baselowerreek

```

This can be then be made into a UI widget that returns the selected surface object.
![surface_selector0](https://user-images.githubusercontent.com/16436291/64949907-83c75800-d87a-11e9-9d51-7be0e50e18b0.gif)

The returned object can then e.g. be used to retrieve, calculate and visualize data through an API or from file based storage:

![surface_selector](https://user-images.githubusercontent.com/16436291/64948863-46fa6180-d878-11e9-8e3f-08b5280d110d.gif)

**Comment by anders-kiaer** [comment]
I think this looks nice 🍰  Some (stupid?) questions to add to the discussion:

 - The reason for having a config file in the first place (and not just present what's on disk) is to limit the amount of data extracted?
 - Should the yaml format have another way of writing dates, e.g. instead of  `20000101` we could write `2001-01-01`, and time delta quantities something like e.g. `(2004-01-01) - (2000-01-01)` instead of `20040101_20000101`.

**Comment by HansKallekleiv** [comment]
Lets try to use fmu-config to generate / replace this config file? @jcrivenaes

**Comment by anders-kiaer** [comment]
The data input configuration for surfaces has now been solved and implemented in #83.
